app/controllers/api/v1/timelines/base_controller.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+class Api::V1::Timelines::BaseController < Api::BaseController
+  after_action :insert_pagination_headers, unless: -> { @statuses.empty? }
+
+  private
+
+  def insert_pagination_headers
+    set_pagination_headers(next_path, prev_path)
+  end
+
+  def pagination_max_id
+    @statuses.last.id
+  end
+
+  def pagination_since_id
+    @statuses.first.id
+  end
+
+  def next_path_params
+    permitted_pagination_params.merge(max_id: pagination_max_id)
+  end
+
+  def prev_path_params
+    permitted_pagination_params.merge(min_id: pagination_since_id)
+  end
+
+  def permitted_pagination_params
+    params
+      .slice(*permitted_params)
+      .permit(*permitted_params)
+  end
```

Comment:
Pushed commit which maybe clarifies that? It's a little awkward because the last method essentially only exists to avoid writing `(*self.class::PERMITTED_PARAMS)` in the first method.

Reply:
I wonder if that would be that bad?